src/vc_util/src/lib.rs

```diff
@@ -94,6 +98,44 @@ pub fn vc_jwt_to_jws(
     Ok(encoder.into_jws(sig))
 }
 
+/// Constructs and returns a JWS (a signed JWT) from the given components.
+/// The given `signing_input` should be a value returned by `vc_signing_input()`
+/// (which already contains a header with canister signatures public key), and
+/// `sig` should be valid canister signature over `signing_input`.
+/// Note: the validity of the signature is not checked.
+pub fn vc_signing_input_to_jws(signing_input: &[u8], sig: &[u8]) -> Result<String, String> {
+    let decoder = Decoder::new();
+    let bytes_with_separators = [signing_input, &[b'.']].concat();
+    let parsed_signing_input = decoder
+        .decode_compact_serialization(&bytes_with_separators, None)
+        .unwrap();
+    let header = parsed_signing_input
+        .protected_header()
+        .expect("internal: failed getting protected header");
+
+    let encoder: CompactJwsEncoder = CompactJwsEncoder::new(parsed_signing_input.claims(), header)
+        .map_err(|e| format!("internal: failed creating JWS encoder: {:?}", e))?;
+    Ok(encoder.into_jws(sig))
+}
+
+/// Extracts the canister signature public key from the given signing_input.
```

Comment:
Maybe expand the comment a bit on what `signing input` is? I.e. a JWT without signature.

Reply:
Done.